tools/SwiftSyntax/SwiftSyntax.swift

```diff
@@ -90,15 +90,20 @@ public final class SyntaxTreeDeserializer {
 public enum SyntaxTreeParser {
   /// Parses the Swift file at the provided URL into a full-fidelity Syntax tree
   /// - Parameter url: The URL you wish to parse.
+  /// - Parameter allowInvalid: If set to `true`, parsing will succeed even if
+  ///             the source code in the given file was invalid. The resulting
+  ///             tree may then contain unknown nodes or nodes with
+  ///             `SourcePresence` `.missing`.
   /// - Returns: A top-level Syntax node representing the contents of the tree,
   ///            if the parse was successful.
   /// - Throws: `ParseError.couldNotFindSwiftc` if `swiftc` could not be
   ///           located, `ParseError.invalidFile` if the file is invalid.
   ///           FIXME: Fill this out with all error cases.
-  public static func parse(_ url: URL) throws -> SourceFileSyntax {
+  public static func parse(_ url: URL, allowInvalid: Bool = false) throws
+      -> SourceFileSyntax {
     let swiftcRunner = try SwiftcRunner(sourceFile: url)
     let result = try swiftcRunner.invoke()
-    guard result.wasSuccessful else {
+    if !result.wasSuccessful && !allowInvalid {
```

Comment:
Syntax errors in source are quite normal, and they should not be treated as exception in `SwiftSyntax` IMO. 
This should catch "invoked but didn't get parsed" error. But once it gets parsed, we might want to have a way to get both parsed syntax tree *and* diagnostics. Until we implement it, `allowInvalid` should be defaulted to `true`. What do you think?

Reply:
Makes total sense to me. I just wanted to keep the current behaviour for now. 

@nkcsgexi Do you have any opinion on this?

Reply:
I think we should treat cases like missing file at the given url or bad encoding as exception. We shouldn't handle invalid syntax at this level since syntax tree can provide interpretation for invalid syntax too.